src/EditorFeatures/Core/Implementation/ChangeSignature/AbstractChangeSignatureCommandHandler.cs

```diff
@@ -126,5 +132,24 @@ private bool ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer, Comma
                 return true;
             }
         }
+
+        private static void ShowError(CannotChangeSignatureReason? reason, IUIThreadOperationContext operationContext, INotificationService notificationService)
+        {
+            switch (reason)
+            {
+                case CannotChangeSignatureReason.DefinedInMetadata:
+                    ShowMessage(FeaturesResources.The_member_is_defined_in_metadata, NotificationSeverity.Error, operationContext, notificationService);
+                    break;
+                case CannotChangeSignatureReason.IncorrectKind:
+                    ShowMessage(FeaturesResources.You_can_only_change_the_signature_of_a_constructor_indexer_method_or_delegate, NotificationSeverity.Error, operationContext, notificationService);
+                    break;
+            }
```

Comment:
so you don't show an error if tehre is no reason?  i don't like this method :)  plz make the reason non-optional and have the caller decide to call it or not as appropriate.